src/libcore/cell.rs

```diff
@@ -378,6 +343,99 @@ impl<T: Copy> From<T> for Cell<T> {
     }
 }
 
+#[unstable(feature = "move_cell", issue = "39264")]
+impl<T> Cell<T> {
+    /// Creates a new `Cell` containing the given value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use std::cell::Cell;
+    ///
+    /// let c = Cell::new(5);
+    /// ```
+    #[stable(feature = "rust1", since = "1.0.0")]
+    #[inline]
+    pub const fn new(value: T) -> Cell<T> {
```

Comment:
Yeah, I'm not sure what the right thing to do here is. There needs to be some way to construct a non-`Copy` `Cell` but perhaps it would be better to add a new function?